src/v/model/fundamental.h

```diff
@@ -123,6 +123,54 @@ class topic : public named_type<ss::sstring, struct model_topic_type> {
 using ns = named_type<ss::sstring, struct model_ns_type>;
 
 using offset = named_type<int64_t, struct model_offset_type>;
+using offset_delta = named_type<int64_t, struct model_offset_delta_type>;
+
+/// \brief conversion from kafka offset to redpanda offset
+inline constexpr model::offset
+operator+(kafka::offset o, model::offset_delta d) {
+    return model::offset{o() + d()};
+}
+
+/// \brief conversion from redpanda offset to kafka offset
+inline constexpr kafka::offset
+operator-(model::offset o, model::offset_delta d) {
+    return kafka::offset{o() - d()};
+}
+
+/// \brief get offset delta from pair of offsets
+inline constexpr model::offset_delta
+operator-(model::offset r, kafka::offset k) {
+    return model::offset_delta{r() - k()};
+}
+
+/// \brief cast to model::offset
+///
+/// The purpose of this function is to mark every place where we converting
+/// from offset-delta to model::offset. This is done in places where the delta
+/// is represetnted as an instance of the model::offset. Once we convert every
+/// delta offset to model::delta_offset we will be able to depricate and remove
+/// this function.
+inline constexpr model::offset offset_cast(model::offset_delta d) {
```

Comment:
:100: really like making this explicit so that we can grep for it when doing subsequent improvements.